### PR TITLE
[backend] Domain observable checker is incorrect (#7620)

### PR DESCRIPTION
**Pull request by SarahBocognano**
### Proposed changes

* Change the regex to also accept dots on subdomain 

### Related issues

* https://github.com/OpenCTI-Platform/opencti/issues/7620

### CODE REVIEW

**Comment by SouadHadjiat** [comment]
unit tests are failing on this PR, also could you add corresponding unit test for the usecase we are fixing ?

**Review by marieflorescontact — COMMENTED** [review]
Tested ok with corrects and incorrects domains and subdomains.
the only case that seems to be not covered by test is a very long subdomain:
"very0long0subdomain0that0exceeds0the0sixty0three0character0limit.mydomain.com"
subdomain are limited to 63 characters to be valid

File: opencti-platform/opencti-graphql/tests/01-unit/utils/syntax-test.ts [hunk 1]
```diff
@@ -12,6 +12,8 @@ describe('Regex Pattern Tests', () => {
     expect('example.com').toMatch(domainChecker);
     expect('sub.example.co.uk').toMatch(domainChecker);
     expect('løveskateboards.com').toMatch(domainChecker);
+    expect('test._mysubdomain.mydomain.com').toMatch(domainChecker);
+    expect('test_mysubdomain.domain.io').toMatch(domainChecker);
     expect('invalid_domain.12_3').not.toMatch(domainChecker);
```
Comment (Archidoit) [inline opencti-platform/opencti-graphql/tests/01-unit/utils/syntax-test.ts:17]: maybe add a test that it's not working if there is a dot in the main name and no dots (exemple: invalid_domain)

**Comment by Archidoit** [comment]
I just suggested to add a little test case. Since everything is ok, I'm gonna approve the PR so you can merge it after adding the test.

**Review by Archidoit — APPROVED** [review]

**Pull request merged**